In [3]:
import pandas as pd
df = pd.read_csv('anime_dataset.csv')
df.head
df.columns

Index(['mal_id', 'title', 'title_english', 'title_japanese', 'type', 'source',
       'episodes', 'status', 'airing', 'aired_from', 'aired_to', 'duration',
       'rating', 'score', 'scored_by', 'rank', 'popularity', 'members',
       'favorites', 'season', 'year', 'studios', 'producers', 'licensors',
       'genres', 'themes', 'demographics', 'synopsis', 'image_url'],
      dtype='object')

#Anime DataSet Summary
I downloaded this data set from Kaggle, which contains information about anime titles, ratungs, popularities, etc. 
Features of this Dataset:
- Anime Title
- Genre
- Rating Score
- Popularity
- # of Episodes
- Studio
- Type of Anime
- Release Year
This dataset stood out to because I enjoy watching all types of anime and I wanted to see what genres and specifically series are the most popular and highly rated. I would find it interesting how different the data is from what social media considers the most popular.

#Questions
1) What are the top 10 highest rated anime?
2) Which anime genre appears most often?
3) What is the average rating of all anime in the dataset?
4) Which studio produced the most anime?
5) Do anime with more episodes tend to have higher ratings?

In [4]:
#Highest Rated Anime
top_rated = df.sort_values(by='score', ascending=False)
top_rated[['title', 'score']].head(10)

,title,score
23037,Sousou no Frieren,9.27
28465,Steel Ball Run: JoJo no Kimyou na Bouken,9.15
3959,Fullmetal Alchemist: Brotherhood,9.11
25957,Chainsaw Man Movie: Reze-hen,9.08
5662,Steins;Gate,9.07
15489,Gintama: The Final,9.05
9871,Gintama°,9.05
14838,Shingeki no Kyojin Season 3 Part 2,9.05
6451,Hunter x Hunter (2011),9.03
27514,One Piece Fan Letter,9.02


The highest rated anime in the dataset is “Sousou no Frieren” with a score of 9.27. Other highly rated anime includec “Fullmetal Alchemist: Brotherhood,” “Steins;Gate,” and “Hunter x Hunter (2011).” Many of the top rated anime are well known series with large fanbases and strong opinionated reviews.

In [5]:
#Most Common Genres
genres = df['genres'].dropna().str.split(',')
all_genres = genres.explode()
all_genres = all_genres.str.strip()
all_genres.value_counts().head(10)

genres
Comedy                      3069
Fantasy                     1504
Hentai                      1325
Slice of Life                938
Drama                        854
Avant Garde                  836
Sci-Fi                       775
Action|Adventure|Fantasy     733
Adventure|Fantasy            623
Comedy|Fantasy               575
Name: count, dtype: int64

The most common genre in the dataset is Comedy, appearing over 3,000 times. Other frequently watched genres include Fantasy, Drama, and Slide of Life. This shows that more lighthearted imaginative genres are more popular when it comes to anime.

In [6]:
#Average Anime Score
df['score'].mean()

6.425589684295193

The average anime score in the dataset is about 6.43. This shows that most of the anime receive a little more positive ratings, with a few very high rated titles that are increasing the overall average.

In [7]:
#Studios With Most Anime Produced
df['studios'].value_counts().head(10)

studios
Toei Animation                    895
Sunrise                           562
J.C.Staff                         418
TMS Entertainment                 380
Shanghai Animation Film Studio    342
Madhouse                          337
Studio Deen                       309
Production I.G                    280
Studio Pierrot                    272
OLM                               265
Name: count, dtype: int64

The studio that appears most frequently in the dataset is Toei Animation, followed by Sunrise and J.C.Staff. These studios have produced a large number of anime titles, which show that they are some of the most active and influential production companies in the industry. This shows that a small number of major studios are responsible for a large portion of anime that has been created.

In [8]:
#Episodes VS Score
df[['episodes', 'score']].corr()

,episodes,score
episodes,1.000000,0.095389
score,0.095389,1.000000


The comparison between the number of episodes and their anime score is about 0.095, which indicates a very weak positive relationship. This suggests that there is litte to nothing connecting how long an anime is and how highly rated it is. Both short and long anime could receive a high or low score, not depending on the amount of episodes they have.

#Conclusion
The analysis of this anime dataset showed some interesting trends arounds its categories and features of anime. The most common genres were Comedy, Fantasy, and Drama, showing that both imagiantion and loghtherated story telling styles are pretty popular. When it came to production, studios like Toei Animation, Sunrise, and J.C.Staff are responsible for a large portion of the dataset. Then the average score in this set is about 6.43, which suggest that most anime is scored more as just mid. For the comparion of the number of episodes and the anime score, it showed that there is not a strong connection between them with it being 0.095. Overall, the data set shows that anime popularity and quality are influenced more by its content, genre, and prodcution factors more than just the simple structual aspects of it like their episode count.